algorithms/kernel/dtrees/gbt/regression/gbt_regression_train_dense_default_impl.i

```diff
@@ -126,8 +166,28 @@ protected:
         const size_t n = this->_dataHelper.data()->getNumberOfRows();
         const algorithmFPType div = algorithmFPType(1.) / algorithmFPType(n);
         val = algorithmFPType(0);
-        for(size_t i = 0; i < n; ++i)
-            val += div*py[i];
+        size_t nThreads = super::numAvailableThreads();
+        if ( nThreads == 0 ) nThreads = 1;
+        if ( n < getThrOptBorder<cpu>(nThreads) || nThreads == 1 )
+        {
+            for(size_t i = 0; i < n; ++i)
+                val += div*py[i];
+        }
+        else
+        {
+            size_t nPerBlock = n / nThreads;
+            size_t nSurplus = n % nThreads;
+            algorithmFPType pvals[nThreads];
```

Comment:
use TArray instead

Reply:
It works only for GCC compiler, it is named as "alloca" - dynamic allocation on stack. For VS compiler - it doesn't work.